pkg/backends/backend.go

```diff
@@ -44,3 +44,14 @@ func NewLoadTest(loadTest *loadTestV1.LoadTest, kubeClientSet kubernetes.Interfa
 	}
 	return nil, fmt.Errorf("load test provider not found: %s", loadTest.Spec.Type)
 }
+
+// BuildLoadTestSpecByBackend returns a valid LoadTestSpec based on backend rules
+func BuildLoadTestSpecByBackend(loadTestType loadTestV1.LoadTestType, overwrite bool, distributedPods int32, testFileStr, testDataStr, envVarsStr string) (loadTestV1.LoadTestSpec, error) {
```

Comment:
We could make this even more generic by doing some sort of Registry.
This would allows Kangal to be extended without even touching kangal repository... eg.:

```go
package main

import (
	"log"

	"github.com/hellofresh/kangal/cmd"
	_ "github.com/my-github-account/my-kangal-new-backend"
)

var version = "0.0.0-dev"

func main() {
	rootCmd := cmd.NewRootCmd(version)

	if err := rootCmd.Execute(); err != nil {
		log.Fatal(err)
	}
}
```

and then:

```go
package my_kangal_new_backend

import (
	"github.com/hellofresh/kangal/pkg/backends"
)

func init() {
	backends.Register(MyNewBackend{})
}
```

What do you think?

Reply:
We might do that once we add other backends, define registry and init rules how to retrieve backend.
I would rather make another ticket for it since this pull request solves issues with not fully described LoadTest spec.